### PR TITLE
Create CJS / UMD bundles

### LINKED ISSUE #5223: Bundles publication strategy
As a follow up to the closed PR #5149 and in particular to the [comments on the downside of the approach taken](https://github.com/angular/angular/pull/5149#issue-115358880) here is a new proposal:

* let's produce 2 UMD bundles:
  * core + RxJS
  * core + http + router + RxJS
* those bundles, being UMD, should be usable via a `<script>` tag and CJS-based tooling (WebPack etc.)
* those bundles won't contain "truly" external dependencies (zone, reflect-metadata, shims) - we are going to publish a separate file with those

@jeffbcross @robwormald @vsavkin - feel free to comment.

//cc.: @IgorMinar @rkirov

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**
Closes #5223

### CODE REVIEW

**Comment by pkozlowski-opensource** [comment]
@vsavkin kindly agreed to review this PR, assigning. Thnx Victor!

File: gulpfile.js [hunk 1]
```diff
@@ -987,6 +987,33 @@ gulp.task('!bundle.js.sfx.dev', ['build.js.dev'], function() {
       });
 });
 
+gulp.task('bundles.js.umd', ['build.js.dev'], function() {
+  var webpack = q.denodeify(require('webpack'));
+
+  var resolveOptions = {
+    root: __dirname + '/dist/js/dev/es5',
```
Comment (vsavkin) [inline gulpfile.js]: Do we only publish dev bundles? What about prod?

Comment (pkozlowski-opensource) [inline gulpfile.js]: This was one of my questions. One could also ask: what about minified vs. unminifed. I don't have a strong opinion here, but with your work on source maps it probably makes sense to have both dev and prod.

My only worry is that is is "exploding" number of bundles.

**Comment by vsavkin** [comment]
>> but with your work on source maps it probably makes sense to have both dev and prod.

Exactly. The prod one should also be minified.

**Comment by vsavkin** [comment]
Add prod bundles. Otherwise, LGTM.

**Comment by pkozlowski-opensource** [comment]
@vsavkin added production bundles as well (minified and not). Not sure if you want to have another look or is this one good to go?

**Comment by pkozlowski-opensource** [comment]
Got LGTM from Victor, let's land it!

**Comment by vsavkin** [comment]
Merged via https://github.com/angular/angular/commit/fa725b4512e280f8fc5dcb8dbf7031a1f0a87f39

**Comment by pkozlowski-opensource** [comment]
Uhhhuuu, can't believe :-) We've got UMD bundles!

**Pull request closed without merging**